jumpy/__init__.py

```diff
@@ -1,39 +1,22 @@
 """Jumpy module."""
 
-from typing import Union
-
-import numpy as onp
-
-dtype = onp.dtype
-pi = onp.pi
-inf = onp.inf
-float32 = onp.float32
-int32 = onp.int32
-
 try:
     import jax.numpy as jnp
 
-    ndarray = Union[onp.ndarray, jnp.ndarray]
-
     is_jax_installed = True
 except ImportError:
-    ndarray = onp.ndarray
-
     is_jax_installed = False
 
 
-from jumpy import lax, ops, random
+from jumpy import core, lax, ops, random
 from jumpy._base_fns import index_update, vmap
+from jumpy._indexing import jparray
 
 __all__ = [
     # === primitives ===
-    "dtype",
-    "pi",
-    "inf",
-    "float32",
-    "int32",
-    "ndarray",
+    "jparray",
     # === sub-modules ===
+    "core",
```

Comment:
We don't need to include `core` if we don't want users to do `jumpy.core.*`. It should be still possible for users to do `from jumpy.core import X`

Reply:
I actually like the `jumpy.core.*` syntax, but if your against I'm fine with removing it!

Reply:
I just realised a major issue, `jax.core.*` already exists, therefore, I think we need to rename the file. 
Also if you want to keep `jumpy.core.*` Im happy to do keep it

Reply:
I find it hard to find a proper alternative. All the logical names seem to already be in use by jax (e.g. `util`). 

Maybe it is not so bad if we keep it in `jumpy.core`. As long as the methods we add to the module either have a unique name (w.r.t `jax.core`), or the methods should have the same signature and have the same function?

Reply:
Ok, we can leave it as it is. Maybe it makes more sense to use `utils` rather than `core`